paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -107,9 +130,74 @@ def paasta_mark_for_deployment(args):
     if service and service.startswith('services-'):
         service = service.split('services-', 1)[1]
     validate_service_name(service, soa_dir=args.soa_dir)
-    return mark_for_deployment(
+    ret = mark_for_deployment(
         git_url=args.git_url,
         deploy_group=deploy_group,
         service=service,
         commit=args.commit,
     )
+    if args.block:
+        try:
+            _log(
+                service=service,
+                line="Waiting for deployment of {0} to {1} complete".format(args.commit, deploy_group),
+                component='deploy',
+                level='event'
+            )
+            wait_for_deployment(args.service, args.deploy_group,
+                                args.commit, args.soa_dir, args.timeout)
+            _log(
+                service=service,
+                line="Deployment of {0} to {1} complete".format(args.commit, deploy_group),
+                component='deploy',
+                level='event'
+            )
+        except TimeoutError:
+            sys.exit(1)
+    return ret
+
+
+def is_instance_deployed(service, instance, git_sha):
+    api = client.get_paasta_api_client()
```

Comment:
D'oh, that's a very good point. I guess I'll need to pull the list of clusters first and then hit the API endpoint for each one...